readthedocs/analytics/signals.py

```diff
@@ -17,16 +18,22 @@ def increase_page_view_count(sender, **kwargs):
 
     project = context['project']
     version = context['version']
-    # footer_response sends an empty path for the index
-    path = context['path'] or '/'
-
-    if project.has_feature(Feature.STORE_PAGEVIEWS):
-        page_view, _ = PageView.objects.get_or_create(
-            project=project,
-            version=version,
-            path=path,
-            date=timezone.now().date(),
-        )
-        PageView.objects.filter(pk=page_view.pk).update(
-            view_count=F('view_count') + 1
-        )
+    origin = kwargs['origin']
+
+    if not origin or not project.has_feature(Feature.STORE_PAGEVIEWS):
+        return
+
+    unresolved = unresolve(origin)
```

Comment:
I'm not sure if this is the best approach right now. I think this is going to add a good number of queries to the footer, which is already our most expensive view. If we can just pull the path off the request, I think that is probably best, but maybe calling `unresolve` is correct, and we should focus on optimizations there?